### PR TITLE
optimize needs to be defined for multiple args

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

**Comment by cpcloud** [comment]
@mrocklin this could use your review when you get a chance

File: blaze/expr/broadcast.py [hunk 1]
```diff
@@ -99,8 +100,8 @@ def scalar_symbols(exprs):
     return scalars
 
 
-Broadcastable = (Arithmetic, Math, Map, Field, DateTime)
-WantToBroadcast = (Arithmetic, Math, Map, DateTime)
+Broadcastable = (Arithmetic, Math, Map, Field, DateTime, USub)
+WantToBroadcast = (Arithmetic, Math, Map, DateTime, USub)
```
Comment (mrocklin) [inline blaze/expr/broadcast.py]: `Not` and `USub` often go together.  Perhaps this should be `UnaryOp`?

Comment (cpcloud) [inline blaze/expr/broadcast.py]: true, then we can remove `Math`, as `UnaryOp` is a superclass of `Math`

Comment (cpcloud) [inline blaze/expr/broadcast.py]: hm but something like `hypot` wouldn't be a unary op nevermind on that last point

**Comment by mrocklin** [comment]
Looks good otherwise.

File: blaze/expr/broadcast.py (hunk 1 above)
Comment (cpcloud) [inline blaze/expr/broadcast.py]: should probably consider decoupling `math` from `unaryop` at some point

Comment (mrocklin) [inline blaze/expr/broadcast.py]: That bit of the class hierarchy could definitely be improved.  Not sure what the right way is.

**Comment by cpcloud** [comment]
turns out `Not` wasn't implemented for pyfunc either

**Comment by cpcloud** [comment]
@mrocklin added unaryop and `Not` implementation to `pyfunc`

any other comments?

**Pull request merged**